cachi2/core/package_managers/gomod.py

```diff
@@ -112,7 +113,13 @@ class Module(NamedTuple):
     @property
     def purl(self) -> str:
         """Get the purl for this module."""
-        return f"pkg:golang/{self.resolved_path}@{self.version}?type=module"
+        purl = PackageURL(
+            type="golang",
+            name=self.resolved_path,
+            version=self.version,
+            qualifiers={"type": "module"},
+        )
+        return purl.to_string()
```

Comment:
IMO worth it, otherwise you would have to mentally validate https://github.com/package-url/purl-spec/blob/master/PURL-SPECIFICATION.rst#character-encoding everywhere

Reply:
Yeah, but in case of Golang, it would only affect us if the module's name had these special characters, which might not even be a valid module name.

On the other hand, cases with special chars could be more common in Pip. If so, it's worth adding it.

Reply:
And will be very common in npm :slightly_smiling_face: 

`@types/react`

Reply:
Gotcha. Let's add it then.